### PR TITLE
Update HTML_Common for PHP7.4+

### PR DESCRIPTION
**Pull request by MisterIcy**
With the coming of PHP8.0, many legacy codebases "broke", for the newest major version contains a lot of changes which are incompatible with older code.

This pull request, contains several changes to modernize the HTML_Common class, of which one is of the greatest importance: the renaming of the class's constructor.

Furthermore a simple workflow was added with linting and basic static analysis, to ensure that the changes work and to facilitate future refactorings.

### CODE REVIEW

File: Common.php [hunk 1]
```diff
@@ -452,7 +450,7 @@ function display()
      * @access  public
      * @static
      */
-    function charset($newCharset = null)
+    public function charset(?string $newCharset = null)
```
Comment (golgote) [inline Common.php]: I think that this function should be static because the charset is global.

File: Common.php [hunk 2]
```diff
@@ -134,12 +134,12 @@ function _getTabs()
...
      * @access   private
      */
-    function _getAttrString($attributes)
+    protected function _getAttrString(array $attributes)
     {
         $strAttr = '';
 
         if (is_array($attributes)) {
-            $charset = HTML_Common::charset();
+            $charset = $this->charset();
```
Comment (golgote) [inline Common.php]: Should use self::charset() instead because html_common is used to know which charset is used globally.

File: Common.php (hunk 1 above)
Comment (MisterIcy) [reply Common.php]: My bad, didn't notice that the method was annotated with `@static`. The method was made static in https://github.com/pear/HTML_Common/pull/3/commits/d5b198ce0da7f6923e1c918039abb6301f6bcaff

File: Common.php (hunk 2 above)
Comment (MisterIcy) [reply Common.php]: Updated in https://github.com/pear/HTML_Common/pull/3/commits/d5b198ce0da7f6923e1c918039abb6301f6bcaff

**Pull request closed without merging**